src/lib/slider/slider.ts

```diff
@@ -151,7 +151,7 @@ export class MdSlider implements AfterContentInit, ControlValueAccessor {
       return;
     }
 
-    this._value = Number(v);
+    this._value = Math.min(Math.max(Number(v), this.min), this.max);
```

Comment:
I'm thinking that it makes more sense to clamp the rendered position of the thumb than the actual value. Having an invalid value is a state that the control should allow, where the user will likely want to show a validation error. 

An easy way to do this would be to have a `getRenderedValue` method that did the clamping.